src/components/presentational/TimeSeriesChart/TimeSeriesChart.tsx

```diff
@@ -226,93 +171,117 @@ export default function TimeSeriesChart(props: TimeSeriesChartProps) {
         dataToDisplay = props.data;
     }
 
+    let yAxisDomain: AxisDomain | undefined = undefined;
+    if (props.options && props.chartType === "Line") {
+        let domainMin = (props.options as MultiSeriesLineChartOptions)?.domainMin;
+        if (domainMin === "Auto") {
+            yAxisDomain = ["auto", "auto"];
+        } else if (domainMin !== undefined) {
+            yAxisDomain = [domainMin, "auto"];
+        }
+    }
+
+    const xAxisTicks = getXAxisTicks();
+
     return <div className="mdhui-time-series-chart" ref={props.innerRef}>
         {props.title &&
             <CardTitle title={props.title}></CardTitle>
         }
         <div className="chart-container">
-            {(!props.chartHasData) &&
-                <div>
-                    {!!dataToDisplay &&
-                        <div className="no-data-label">No Data</div>
+            {!props.chartHasData && !!dataToDisplay && <div className="no-data-label">No Data</div>}
+            {!props.chartHasData && !dataToDisplay && <LoadingIndicator />}
+            <ResponsiveContainer width="100%" height={150}>
+                <ComposedChart data={dataToDisplay} syncId={props.syncId}>
+                    {props.chartHasData && props.tooltip &&
+                        <Tooltip wrapperStyle={{ outline: "none" }} active content={<props.tooltip />} />
                     }
-                    {!dataToDisplay &&
-                        <LoadingIndicator />
+                    <CartesianGrid vertical={props.chartType !== "Bar"} strokeDasharray="2 4" />
+                    <YAxis
+                        tickFormatter={tickFormatter}
+                        axisLine={false}
+                        interval={0}
+                        tickLine={false}
+                        width={32}
+                        domain={yAxisDomain}
+                    />
+                    <XAxis
+                        id="myXAxis"
+                        domain={[xAxisTicks![0], xAxisTicks![xAxisTicks!.length - 1]]}
+                        padding={props.chartType === 'Bar' ? 'gap' : { left: 0, right: 0 }}
+                        tick={DayTick}
+                        scale={'time'}
+                        type={'number'}
+                        axisLine={false}
+                        dataKey="timestamp"
+                        tickMargin={0}
+                        minTickGap={0}
+                        tickLine={false}
+                        ticks={xAxisTicks}
+                        includeHidden
+                        interval={0}
+                    />
+                    {props.children}
```

Comment:
Here is where the overlay children are rendered.  This is the reason for switching to the `ComposedChart`.

The children can be things such as references lines, bars, lines, areas, etc.